search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -188,31 +197,47 @@
   (let [bucket (first (get elastic-aggregations :buckets))
         value (extract-value-from-bucket bucket field)
-        sub-facets (recursive-parse-fn value elastic-aggregations)
+        sub-facets (recursive-parse-fn (rest field-hierarchy) value elastic-aggregations)
         count (reduce + (map :count (:children sub-facets)))
         sub-facets (when (seq (:children sub-facets))
                      (update sub-facets :children
                              #(sort-by :title util/compare-natural-strings %)))
         children-values-to-remove (find-applied-children sub-facets field-hierarchy false)
         has-siblings? (has-siblings-fn value)
         links (generate-links-fn value has-siblings? children-values-to-remove)]
-    [(v2h/generate-hierarchical-filter-node value count links sub-facets)]))
+    [(v2h/generate-hierarchical-filter-node value count links sub-facets field)]))
 
-(defn- process-bucket-for-hierarchical-field
-  "Takes an elasticsearch bucket for a hierarchical field and returns a V2 facets node for that
-  bucket."
-  [bucket field field-hierarchy recursive-parse-fn has-siblings-fn generate-links-fn]
-  (let [value (extract-value-from-bucket bucket field)
-        count (get-in bucket [:coll-count :doc_count] (:doc_count bucket))
-        sub-facets (recursive-parse-fn value bucket)
-        ;; Sort alphabetically
-        sub-facets (when (seq (:children sub-facets))
+(defn- finish-bucket-for-hierarchical-field
+  "Called from process-bucket-for-hierarchical-field to finish up creating
+  the V2 facet node for a specific bucket and returns the V2 facet node."
+  [sub-facets field-hierarchy has-siblings-fn generate-links-fn value count field]
+  (let [sub-facets (when (seq (:children sub-facets)) ; Sort alphabetically
                      (update sub-facets :children
                              #(sort-by :title util/compare-natural-strings %)))
         children-values-to-remove (find-applied-children sub-facets field-hierarchy false)
         has-siblings? (has-siblings-fn value)
         links (generate-links-fn value has-siblings? children-values-to-remove)]
-    (v2h/generate-hierarchical-filter-node value count links sub-facets)))
+    (v2h/generate-hierarchical-filter-node value count links sub-facets field)))
+
+(defn- process-bucket-for-hierarchical-field
+  "Takes an elasticsearch bucket for a hierarchical field and returns a V2 facets node for that
+  bucket."
+  [bucket field field-hierarchy recursive-parse-fn has-siblings-fn generate-links-fn function-params]
+  (let [value (extract-value-from-bucket bucket field)
+        count (get-in bucket [:coll-count :doc_count] (:doc_count bucket))
+        subfield (if (= (:base-field function-params) :science-keywords-h)
+                   :detailed-variable
+                   :short-name)
+        sub-facets (recursive-parse-fn (rest field-hierarchy) value bucket)]
+    (if sub-facets
```

Comment:
can there be fewer nested if statements or a comment describing what is going on here?

Reply:
I can (cond) and it will look a little better, I don't think that will make it clearer.  I can put in some comments.

Reply:
Updated to use (cond) and added a lot of comments.